packages/playground/src/weblets/tf_nextcloud.vue

```diff
@@ -28,199 +29,222 @@
       </SelectFarmManager>
     </form-validator>
+
     <template #footer-actions>
-      <v-btn color="primary" variant="tonal" :disabled="!valid" @click="deploy">Deploy</v-btn>
+      <v-btn
+        color="primary"
+        variant="tonal"
+        @click="deploy(domainNameCmp?.domain, domainNameCmp?.customDomain)"
+        :disabled="!valid"
+      >
+        Deploy
+      </v-btn>
     </template>
   </weblet-layout>
 </template>
 
 <script lang="ts" setup>
+import type { GridClient } from "@threefold/grid_client";
 import { computed, type Ref, ref } from "vue";
 
-import Network from "../components/networks.vue";
-import SelectSolutionFlavor from "../components/select_solution_flavor.vue";
 import { useLayout } from "../components/weblet_layout.vue";
 import { useProfileManager } from "../stores";
-import { type Farm, type Flist, ProjectName } from "../types";
-import { deployVM, type Disk, type Env } from "../utils/deploy_vm";
+import type { Farm, Flist, GatewayNode, solutionFlavor as SolutionFlavor } from "../types";
+import { ProjectName } from "../types";
+import { deployVM } from "../utils/deploy_vm";
+import { deployGatewayName, getSubdomain, rollbackDeployment } from "../utils/gateway";
 import { getGrid } from "../utils/grid";
+import { normalizeError } from "../utils/helpers";
 import { generateName } from "../utils/strings";
 
 const layout = useLayout();
 const valid = ref(false);
-const domain = ref("");
 const profileManager = useProfileManager();
 
+const name = ref(generateName({ prefix: "nc" }));
+const solution = ref() as Ref<SolutionFlavor>;
+const farm = ref() as Ref<Farm>;
 const flist: Flist = {
   value: "https://hub.grid.tf/tf-official-apps/threefoldtech-nextcloudaio-latest.flist",
   entryPoint: "/sbin/zinit init",
 };
-
-const name = ref(generateName({ prefix: "nc" }));
-const ipv4 = ref(true);
-const planetary = ref(true);
-const wireguard = ref(true);
-const farm = ref() as Ref<Farm>;
-const network = ref();
 const dedicated = ref(false);
 const certified = ref(false);
 const selectedNode = ref() as Ref<INode>;
-const rootFilesystemSize = 40;
-
-async function deploy() {
+const ipv4 = ref(false);
+const loadingFarm = ref(false);
+const domainNameCmp = ref();
+const rootFilesystemSize = computed(() => rootFs(solution.value?.cpu ?? 0, solution.value?.memory ?? 0));
+function finalize(deployment: any) {
+  layout.value.reloadDeploymentsList();
+  layout.value.setStatus(
+    "success",
+    "Successfully deployed a Nextcloud instance. Under Actions, click on the button Nextcloud Setup to set up Nextcloud. After installation, you can access the Nextcloud instance by clicking on the Open Nextcloud button or navigating to your Nextcloud domain.",
+  );
+  layout.value.openDialog(deployment, deploymentListEnvironments.nextcloud);
+}
+async function deploy(gatewayName: GatewayNode, customDomain: boolean) {
   layout.value.setStatus("deploy");
 
   const projectName = ProjectName.Nextcloud.toLowerCase();
```

Comment:
```suggestion
  const projectName = ProjectName.Nextcloud.toLowerCase() + "/" + name.value;
```
Please Add /{name} as suffix to projectName as we changing the way we storing all machines to include gateways